### DISCUSSION TITLE
Sliced Prediction question: How to prevent detection in bigger area than slice

### DISCUSSION THREAD

**Discussion by Makarand-F4F** [obss/sahi#1081] [Q&A] [answered]
Hello, 
I am very new to Sahi and have just recently started using it with a trained detectron-2 model. I have specified the sliced predictions and the model as below. The saved model that I am specifying was trained on 200 x 200 pixel images and I am using the same slice_height and slice_width for prediction.

```
detection_model = AutoDetectionModel.from_pretrained(
    model_type='detectron2',
    model_path="path",
    config_path="another path",
    confidence_threshold=conf_thresh,
    # image_size=640, # this I have commented out
    device = 'cuda:0',) 
```

```
  result = get_sliced_prediction(
      image_path,
      detection_model,
      slice_height = 200,
      slice_width = 200,
      overlap_height_ratio = 0.00,
      overlap_width_ratio = 0.00,)
```

Now, when I do the predictions, I expect the model to perform the object detection on each of the slice and then combine the results together but sometimes, I see a prediction that spans across multiple slices and using a much bigger tile. For examples, see the images below. First image shows correct small rectangles detected in each 200 x 200 pixel slice.

In this example image, you can see my detections as expected
![image](https://github.com/user-attachments/assets/4b38b6d6-872c-40ea-8e96-aad0385c2ce5)

In this image, you can see a much larger bounding box around an object that has been detected across much larger area than 200 x 200. How do I prevent this?
![image](https://github.com/user-attachments/assets/650154a8-5e40-4e69-afd6-df313ee7f1f0)

**Comment by fcakyon** [comment, marked as answer]
@Makarand-F4F you can achieve it by setting `perform_standard_pred=False`:

https://github.com/obss/sahi/blob/3c8f593e1d23080fad7fe6c9ba1d33cd26e271d2/sahi/predict.py#L132

**Reply by Makarand-F4F** [reply]
Thank you @fcakyon 
I read this a long time ago but forgot to thank you!

**Reply by fcakyon** [reply]
I'm glad you find it helpful @Makarand-F4F ! 🎉
